prefect_kubernetes/jobs.py

```diff
@@ -0,0 +1,304 @@
+"""Module to define tasks for interacting with Kubernetes jobs."""
+
+from typing import Any, Dict, Optional
+
+from kubernetes import client
+from kubernetes.client.models import V1Job, V1JobList, V1Status
+from prefect import task
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+
+from prefect_kubernetes.credentials import KubernetesCredentials
+
+
+@task
+async def create_namespaced_job(
+    body: Dict,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1Job:
+    """Task for creating a namespaced Kubernetes job.
+
+    Args:
+        body: A dictionary representing a Kubernetes V1Job specification.
+        namespace: The Kubernetes namespace to create this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        **kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+
+    Returns:
+        A Kubernetes `V1Job` object.
+
+    Example:
+        Create a job in the default namespace:
+        ```python
+        from prefect import flow
+        from prefect_kubernetes.jobs import create_namespaced_job
+
+        @flow
+        def kubernetes_orchestrator():
+            v1_job_metadata = create_namespaced_job(
+                body={"metadata": {"name": "test-job"}},
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+            )
+        ```
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    return await run_sync_in_worker_thread(
+        api_client.create_namespaced_job, namespace=namespace, body=body, **kube_kwargs
+    )
+
+
+@task
+async def delete_namespaced_job(
+    job_name: str,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict[str, Any]] = None,
+    delete_option_kwargs: Optional[Dict] = None,
```

Comment:
I would only keep one of these.

```suggestion
    **kube_kwargs: Optional[Dict[str, Any]],
```